### PR TITLE
0.7 compat fixes

### PR DESCRIPTION
**Pull request by staticfloat**
Quash those bugs like it's 2018

### CODE REVIEW

File: test/REQUIRE [hunk 1]
```diff
@@ -0,0 +1 @@
+Compat
```
Comment (tkelman) [inline test/REQUIRE]: doing this is a test dependency will mask the fact that codeunits and uninitialized are used in src but not present in 0.6

Comment (staticfloat) [reply test/REQUIRE]: Doy, of course.  Thanks.

File: REQUIRE [hunk 2]
```diff
@@ -1 +1,2 @@
 julia 0.6
+Compat
```
Comment (tkelman) [inline REQUIRE]: prior to 0.38 won't support uninitialized https://github.com/JuliaLang/Compat.jl/commit/c72b467cf9693308fa99eddba9abfcc378f25147

Comment (staticfloat) [reply REQUIRE]: Thanks, I'll add that into https://github.com/staticfloat/SHA.jl/pull/52

Comment (tkelman) [reply REQUIRE]: did you already tag this with a too-loose requirement? are copyto and GC even in a Compat release yet?

Comment (staticfloat) [reply REQUIRE]: `copyto!()` and `GC` aren't merged into `master` yet, so they're not tagged.  I did already tag this with a too-loose requirement.  Unfortunately CI doesn't catch things like this.  :)

Comment (tkelman) [reply REQUIRE]: You can amend it in metadata. Also see RequirementVersions.jl. You folks have gotten worse lately about merging and releasing changes to widely-used packages without even waiting for anyone else to review. I'm glad I don't actually rely on Julia any more but it's unfortunate for those who do.

**Pull request merged**